### DISCUSSION TITLE
What does true for repeatable responses imply regarding etag use on Api Listener in latest frankdoc?

### DISCUSSION THREAD

**Discussion by Laurens-makel** [frankframework/frankframework#5119] [Q&A] [answered]
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
The FrankDoc states the following:

![image](https://github.com/ibissource/iaf/assets/37303445/edea65e8-79e7-40ac-8a58-6183d060251e)

What is actually implied here by 'repeatable responses'?

**Comment by nielsm5** [comment]
It's a bit tricky to explain, but in simple terms, things like a String (character data) or byte[] (binary data) that are preserved in memory are repeatable. _They can be read multiple times._ Things like a Reader or InputStream are not repeatable.
In the latest versions of the framework, large non-repeatable messages may be preserved to disk, which will make them repeatable. 

The reason this behaviour has changed, is that the response (when it's a stream) must be full read in order to calculate the ETAG hash. This may have a negative impact on memory usage. You can read more in #3452.

I doubt that issue still exists with the introduction of #4759, which has been backported to the 7.8.1.

**Reply by Laurens-makel** [reply]
Thanks for the clarification!

Regarding storing large non repeatable messages in files, on disk sounds like the repeatable response is bound the node that initially processed the message.

What would happen if the consumer hits another node, which does not have this file on disk, with a valid etag?

**Reply by nielsm5** [reply]
The 'file-on-disk' is a temporary file and will be removed after it's been read or the pipeline closes. This is part of the (new) 'natural' behaviour of the Franks. The etag is calculated during that process, once calculated it's stored in a cache which can be accessed by multiple nodes when you use memcached.

**Reply by nielsm5** [reply, marked as answer]
I've changed the default behavior to false in #5317. Changed the documentation as well in order to avoid more confusion 😃 .
